Topics/Tech_Stacks/redux.md

```diff
@@ -25,6 +26,17 @@ This tutorial provides a comprehensive guide to Redux, a popular state managemen
 
 Redux is a predictable state container for JavaScript applications. It's used for managing the state of your app in a more predictable and easier-to-debug manner. Redux maintains the state of an entire application in a single immutable state tree, which can't be changed directly. When changes are made, a new object is created and merged with the state.
 
+## Why Do We Use Redux?
```

Comment:
I see,  Redux maye be is better suited  for large scale applications. Could you explain what efforts Redux make to make the state changes maintainable and easy to understand?

Reply:
Redux make the state changes maintainable and easy to understand in two ways:
+ Redux maintains the entire state of your application in a single store. This centralized approach ensures that the state is consistent across the app and makes it easier to track and manage state changes.
+ In Redux, the state is immutable, meaning it cannot be changed directly. Instead, any state change produces a new state object. This immutability simplifies tracking changes over time and debugging, as each state in the application's lifecycle can be inspected and understood in isolation.

Reply:
That makes sense a lot. So, it basically keep everything is single immutable and trackable state, which enhances the tracking of states across the whole app.